### PR TITLE
Fix a bug in required feature checks

### PR DESCRIPTION
**Pull request by TheBlueMatt**
I'm sure we'll hit more such things when we do compat testing of the MPP stuff, but I think this one is Somewhat Obvious (tm), plus a few really trivial tests for features. When we do compat testing we should make sure to extract serialized features from other nodes and add those as tests. Based on the ongoing stack.

### CODE REVIEW

File: lightning/src/ln/features.rs [hunk 1]
```diff
@@ -284,3 +284,38 @@ impl<R: ::std::io::Read, T: sealed::Context> Readable<R> for Features<T> {
...
+		assert!(!InitFeatures::supported().supports_unknown_bits());
+		assert!(!NodeFeatures::supported().requires_unknown_bits());
+		assert!(!NodeFeatures::supported().supports_unknown_bits());
+
+		assert!(InitFeatures::supported().supports_upfront_shutdown_script());
+		assert!(NodeFeatures::supported().supports_upfront_shutdown_script());
+
+		assert!(InitFeatures::supported().supports_data_loss_protect());
+		assert!(NodeFeatures::supported().supports_data_loss_protect());
+
```
Comment (ariard) [inline lightning/src/ln/features.rs]: You can check for `variable_length_onion` now

File: lightning/src/ln/features.rs [hunk 2]
```diff
@@ -187,7 +187,7 @@ impl<T: sealed::Context> Features<T> {
 	pub(crate) fn requires_unknown_bits(&self) -> bool {
 		self.flags.iter().enumerate().any(|(idx, &byte)| {
 			(match idx {
-				0 => (byte & 0b00010100),
+				0 => (byte & 0b01000100),
```
Comment (ariard) [inline lightning/src/ln/features.rs]: Oh can you add comment like I was suggesting here https://github.com/rust-bitcoin/rust-lightning/pull/434#discussion_r377866024

**Comment by TheBlueMatt** [comment]
I believe I resolved the two concerns, @ariard .

**Comment by ariard** [comment]
ACK 4ac9ed2

**Pull request merged**